x-pack/agent/pkg/fleetapi/checkin_cmd.go

```diff
@@ -53,17 +55,16 @@ func (e *CheckinResponse) Validate() error {
 
 // CheckinCmd is a fleet API command.
 type CheckinCmd struct {
-	client      clienter
-	checkinPath string
+	client           clienter
+	agentIDRetriever func() string
 }
 
 // NewCheckinCmd creates a new api command.
-func NewCheckinCmd(agentID string, client clienter) *CheckinCmd {
-	const p = "/api/fleet/agents/%s/checkin"
+func NewCheckinCmd(agentID func() string, client clienter) *CheckinCmd {
```

Comment:
@michalpristas What if we try to make it more future proof and we have an _AgentInfo_ struct that expose the `ID()` method, this will effectively limit required changes if we have more internal data to pass around the code?